### ISSUE TITLE
v1.8.0 blocking test list

### ISSUE THREAD

**Issue by spiffxp** [kubernetes/kubernetes#53018]
The 1.8 release team would like to identify the list of tests we are using for go / no-go signal.  This umbrella issue is to gain consensus on (and link to PR's that help with):

- our decision to continue the trend of expecting tests to pass 3 times in a row for go / no-go signal, except for...
- tests we will be willing to relax this expectation (eg: upgrade tests, flaky tests, and long tests)
- tests we believe should be removed from the release-1.8-blocking dashboard
- upgrade tests we are willing to explain away as known failures

---

[release-1.8-blocking](https://k8s-testgrid.appspot.com/release-1.8-blocking)

- [build-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#build-1.8)
- [verify-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#verify-1.8)
- [test-go-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#test-go-1.8)
- [gce-kubeadm-1.7-on-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gce-kubeadm-1.7-on-1.8)
- [periodic-gce-kubeadm-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#periodic-gce-kubeadm-1.8)
- [gci-gce-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gce-1.8)
- [gci-gce-slow-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gce-slow-1.8)
  - flaky suite, unique flakes, it may be difficult to achieve three passes in a row
- [gci-gce-serial-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gce-serial-1.8)
  - long suite, takes ~15h to achieve 3 consecutive passes
- [gci-gce-alpha-features-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gce-alpha-features-1.8)
- [gci-gce-scalability-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gce-scalability-1.8)
- [gci-gce-ingress-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gce-ingress-1.8)
- [gci-gce-reboot-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gce-reboot-1.8)
- [gci-gke-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gke-1.8)
  - possible flaky suite with unique flakes (two frequent flakes)
  - ~fixed non-blocking most frequent flake: [sig-api-machinery] Aggregator Should be able to support the 1.7 Sample API Server using the current Aggregator ([#50945](https://github.com/kubernetes/kubernetes/issues/50945))  (pr: [#53030](https://github.com/kubernetes/kubernetes/pull/53030))~
  - next most frequent flake: [sig-apps] Job should run a job to completion when tasks sometimes fail and are not locally restarted
- [gci-gke-slow-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gke-slow-1.8)
- [gci-gke-serial-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gke-serial-1.8)
  - long suite, takes ~15h to achieve 3 consecutive passes
- [gci-gke-ingress-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gke-ingress-1.8)
- [gci-gke-reboot-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gci-gke-reboot-1.8)
- [kops-aws-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#kops-aws-1.8)
- [federation-build-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#federation-build-1.8)
- [federation-test-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#federation-test-1.8)
- [kubelet-1.8](https://k8s-testgrid.appspot.com/release-1.8-blocking#kubelet-1.8)
  - possibly flaky suite (two flakes)
  - most frequent flake: \[k8s.io\] Security Context when creating a pod in the host network namespace should listen on same port in the host network containers ([#53091](https://github.com/kubernetes/kubernetes/issues/53091))
  - low frequency flake: \[k8s.io\] Pods should be submitted and removed [Conformance] 
- [gce-gpu-1-8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gce-gpu-1-8)
- [gke-gpu-1-8](https://k8s-testgrid.appspot.com/release-1.8-blocking#gke-gpu-1-8)

---

[release-1.8-upgrade-skew](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew)

- [gke-gci-1.6-gci-1.8-upgrade-master](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-gci-1.6-gci-1.8-upgrade-master)
  - ~fail: [k8s.io] DNS should provide DNS for ExternalName services ([#50274](https://github.com/kubernetes/kubernetes/issues/50274)) (pr: [#53015](https://github.com/kubernetes/kubernetes/issues/53015))~
  - ~non-blocking wont-fix fail: [k8s.io] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should not deadlock when a pod's predecessor fails ([#52808](https://github.com/kubernetes/kubernetes/issues/52808))~
  - ~non-blocking wont-fix fail : [k8s.io] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should provide basic identity ([#52808](https://github.com/kubernetes/kubernetes/issues/52808))~
- [gke-gci-1.6-gci-1.8-upgrade-cluster](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-gci-1.6-gci-1.8-upgrade-cluster)
  - ~fail: [k8s.io] DNS should provide DNS for ExternalName services ([#50274](https://github.com/kubernetes/kubernetes/issues/50274))  (pr: [#53015](https://github.com/kubernetes/kubernetes/issues/53015))~
  - ~non-blocking wont-fix fail: [k8s.io] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should not deadlock when a pod's predecessor fails ([#52808](https://github.com/kubernetes/kubernetes/issues/52808))~
  - ~non-blocking wont-fix fail: [k8s.io] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should provide basic identity ([#52808](https://github.com/kubernetes/kubernetes/issues/52808))~
- [gke-gci-1.6-gci-1.8-upgrade-cluster-new](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-gci-1.6-gci-1.8-upgrade-cluster-new)
  - ~non-blocking frequent flake: [sig-api-machinery] Aggregator Should be able to support the 1.7 Sample API Server using the current Aggregator (pr: [#53030](https://github.com/kubernetes/kubernetes/pull/53030))~
- [gke-gci-1.7-gci-1.8-upgrade-master](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-gci-1.7-gci-1.8-upgrade-master)
  - long suite: ~36h to achieve 3 consecutive passes
  - ~fail: [k8s.io] DNS should provide DNS for ExternalName services ([#50274](https://github.com/kubernetes/kubernetes/issues/50274)) (pr: [#52875](https://github.com/kubernetes/kubernetes/pull/52875))~
- [gke-gci-1.7-gci-1.8-upgrade-cluster](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-gci-1.7-gci-1.8-upgrade-cluster)
  - long suite: ~36h to achieve 3 consecutive passes
  - fail: [k8s.io] DNS should provide DNS for ExternalName services ([#50274](https://github.com/kubernetes/kubernetes/issues/50274)) (pr: [#52875](https://github.com/kubernetes/kubernetes/pull/52875))
- [gke-gci-1.7-gci-1.8-upgrade-cluster-new](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-gci-1.7-gci-1.8-upgrade-cluster-new)
  - long suite: ~36h to achieve 3 consecutive passes
- [gke-cvm-1-6-gci-1-8-upgrade-master](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-cvm-1-6-gci-1-8-upgrade-master)
  - BeforeSuite * ???
  - ~fail: [k8s.io] DNS should provide DNS for ExternalName services ([#50274](https://github.com/kubernetes/kubernetes/issues/50274)) (pr:  #[#53015](https://github.com/kubernetes/kubernetes/issues/53015))~
  - ~non-blocking wont-fix fail: [k8s.io] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should not deadlock when a pod's predecessor fails ([#52808](https://github.com/kubernetes/kubernetes/issues/52808))~
  - ~non-blocking wont-fix fail: [k8s.io] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should provide basic identity ([#52808](https://github.com/kubernetes/kubernetes/issues/52808))~
- [gke-cvm-1-6-gci-1-8-upgrade-cluster](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-cvm-1-6-gci-1-8-upgrade-cluster)
  - Overall failing ???
  - ~fail: [k8s.io] DNS should provide DNS for ExternalName services ([#50274](https://github.com/kubernetes/kubernetes/issues/50274)) (pr:  #[#53015](https://github.com/kubernetes/kubernetes/issues/53015))~
  - ~non-blocking wont-fix fail: [k8s.io] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should not deadlock when a pod's predecessor fails ([#52808](https://github.com/kubernetes/kubernetes/issues/52808))~
  - ~non-blocking wont-fix faill: [k8s.io] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should provide basic identity ([#52808](https://github.com/kubernetes/kubernetes/issues/52808))~
- [gke-cvm-1-6-gci-1-8-upgrade-cluster-new](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-cvm-1-6-gci-1-8-upgrade-cluster-new)
  - flaky, unlikely to see 3 passes in a row
  - most flakes: [sig-cluster-lifecycle] Upgrade [Feature:Upgrade] cluster upgrade should maintain a functioning cluster [Feature:ClusterUpgrade] (https://github.com/kubernetes/kubernetes/issues/53145)
- [gke-cvm-1-7-gci-1-8-upgrade-master](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-cvm-1-7-gci-1-8-upgrade-master)
  - long suite: ~36h to achieve three consecutive passes 
  - failing (unable to dump cluster logs)
- [gke-cvm-1.7-gci-1.8-upgrade-cluster](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-cvm-1.7-gci-1.8-upgrade-cluster)
  - long suite: ~36h to achieve three consecutive passes 
  - failing (unable to dump cluster logs)
- [gke-cvm-1.7-gci-1.8-upgrade-cluster-new](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-cvm-1.7-gci-1.8-upgrade-cluster-new)
  - long suite: ~36h to achieve three consecutive passes 
  - failing (unable to dump cluster logs)
- [gce-1.7-1.8-gci-kubectl-skew](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gce-1.7-1.8-gci-kubectl-skew)
- [gce-1.8-1.7-gci-kubectl-skew](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gce-1.8-1.7-gci-kubectl-skew)
- [gke-1.8-1.7-gci-kubectl-skew](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-1.8-1.7-gci-kubectl-skew)
- [gke-1.7-1.8-gci-kubectl-skew](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-1.7-1.8-gci-kubectl-skew)
- [gce-1.7-1.8-gci-kubectl-skew-serial](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gce-1.7-1.8-gci-kubectl-skew-serial)
- [gce-1.8-1.7-gci-kubectl-skew-serial](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gce-1.8-1.7-gci-kubectl-skew-serial)
- [gke-1.8-1.7-gci-kubectl-skew-serial](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-1.8-1.7-gci-kubectl-skew-serial)
- [gke-1.7-1.8-gci-kubectl-skew-serial](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-1.7-1.8-gci-kubectl-skew-serial)
- [gce-1.7-1.8-upgrade-master](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gce-1.7-1.8-upgrade-master)
  - long suite: ~36h to achieve three consecutive passes 
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] Pods should return to running and ready state after network partition is healed All pods on the unreachable node should be marked as NotReady upon the node turn NotReady AND all pods should be mark back to Ready when the node get back to Ready before pod eviction timeout (#47820)~
- [gce-1.7-1.8-upgrade-cluster](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gce-1.7-1.8-upgrade-cluster)
  - long suite: ~36h to achieve three consecutive passes
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] Pods should return to running and ready state after network partition is healed All pods on the unreachable node should be marked as NotReady upon the node turn NotReady AND all pods should be mark back to Ready when the node get back to Ready before pod eviction timeout (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [Job] should create new pods when node is partitioned (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [ReplicationController] should eagerly create replacement pod during network partition when termination grace is non-zero (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [ReplicationController] should recreate pods scheduled on the unreachable node AND allow scheduling of pods on a node after it rejoins the cluster (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [StatefulSet] should not reschedule stateful pods if there is a network partition [Slow] [Disruptive] (#47820)~
- [gce-1.7-1.8-upgrade-cluster-new](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gce-1.7-1.8-upgrade-cluster-new)
  - long suite: ~36h to achieve three consecutive passes
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] Pods should return to running and ready state after network partition is healed All pods on the unreachable node should be marked as NotReady upon the node turn NotReady AND all pods should be mark back to Ready when the node get back to Ready before pod eviction timeout (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [Job] should create new pods when node is partitioned (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [ReplicationController] should eagerly create replacement pod during network partition when termination grace is non-zero (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [ReplicationController] should recreate pods scheduled on the unreachable node AND allow scheduling of pods on a node after it rejoins the cluster (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [StatefulSet] should not reschedule stateful pods if there is a network partition [Slow] [Disruptive] (#47820)~
- [gce-1.8-1.7-downgrade-cluster](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gce-1.8-1.7-downgrade-cluster)
  - long suite: ~21h to achieve three consecutive passes
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] Pods should return to running and ready state after network partition is healed All pods on the unreachable node should be marked as NotReady upon the node turn NotReady AND all pods should be mark back to Ready when the node get back to Ready before pod eviction timeout (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [Job] should create new pods when node is partitioned (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [ReplicationController] should eagerly create replacement pod during network partition when termination grace is non-zero (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [ReplicationController] should recreate pods scheduled on the unreachable node AND allow scheduling of pods on a node after it rejoins the cluster (#47820)~
  - ~non-release-blocker fail: [k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] [StatefulSet] should not reschedule stateful pods if there is a network partition [Slow] [Disruptive] (#47820)~
- [gce-1.8-1.7-downgrade-cluster-parallel](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gce-1.8-1.7-downgrade-cluster-parallel)
- [gke-1.8-1.7-downgrade-cluster](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-1.8-1.7-downgrade-cluster)
  - long suite: ~21h to achieve three consecutive passes
  - fail: [k8s.io] SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run [Conformance] ([#53066](https://github.com/kubernetes/kubernetes/issues/53066))
  - flaky: hpa-upgrade
- [gke-1.8-1.7-downgrade-cluster-parallel](https://k8s-testgrid.appspot.com/release-1.8-upgrade-skew#gke-1.8-1.7-downgrade-cluster-parallel)
  - flaky: hpa-upgrade

---

[release-master-blocking](https://k8s-testgrid.appspot.com/release-master-blocking)

- [gce-scale-correctness](https://k8s-testgrid.appspot.com/release-master-blocking#gce-scale-correctness)
  - long suite, use solely as information to make sure no new scale issues have appeared
- [gce-scale-performance](https://k8s-testgrid.appspot.com/release-master-blocking#gce-scale-performance)
  - long suite, use solely as information to help decide whether API call latencies are still out of SLO bounds on large clusters ([#51899](https://github.com/kubernetes/kubernetes/issues/51899))

/sig release
/priority critical-urgent
/kind cleanup

**Comment by spiffxp** [comment]
@kubernetes/sig-cluster-lifecycle-test-failures @krzyzacy @luxas can you mention this issue in a PR to move the kubeadm jobs?

@kubernetes/sig-federation-test-failures @madhusudancs @csbell if you've got specific issues you're working to make sure the federation jobs are running again, please link here

**Comment by pipejakob** [comment]
> gce-kubeadm-1.8
> would like to remove, there has been no equivalent on release-master-blocking

I see two completely green runs on 9/22, followed by one that is still running. Is this an argument for adding the test to release-master-blocking? I thought we had gotten messaging from @jdumars during a SIG Cluster Lifecycle to keep the kubeadm e2e tests out of `release-master-blocking`, but that adding them to `1.8-blocking` was still fine.

> gce-kubeadm-1.7-on-1.8
> would like to remove, there has been no equivalent on release-master-blocking
> BeforeSuite continues to show as failed despite two most recent green runs

I similarly see two green runs on 9/22, with the newest job still running. I think testgrid is a little misleading for this test, since it hasn't failed `BeforeSuite` since 9/15, but it still appears red on the left. Otherwise, this looks healthy to me.

> periodic-gce-kubeadm-1.8
> would like to remove, there has been no equivalent on release-master-blocking
> last 3 runs stuck in R state

This job looks incredibly green to me since 9/23, with only an occasional flaky failure. Nothing stuck in running.

> periodic-kubernetes-e2e-kubeadm-gce-upgrade-1-7-1-8
> would like to remove, there has been no equivalent on release-master-blocking
> has been failing since its introduction 9/22

No arguments here. We are still working on getting this test green, but might need to resort to manual upgrade testing for kubeadm. I don't think this is blocking the release.

**Comment by luxas** [comment]
@spiffxp as discussed on Slack, here's the breakdown of the jobs:
https://k8s-testgrid.appspot.com/sig-cluster-lifecycle#periodic-kubeadm-gce-1.8 green, exactly the same as https://k8s-testgrid.appspot.com/sig-cluster-lifecycle#kubeadm-gce-1.8 which also is green

https://k8s-testgrid.appspot.com/sig-cluster-lifecycle#kubeadm-gce-1.7-on-1.8 tests deploying a v1.7 cluster with v1.8 kubeadm CLI, this job is also green.

https://k8s-testgrid.appspot.com/sig-cluster-lifecycle#kubeadm-gce-upgrade-1.7-1.8 will probably get back on its feet, was just renamed, put into the dashboard and suffered from a small test-infra misconfig.
Now also moved into the separate dashboard for upgrades.

It was an oversight that these weren't master blocking, this is now fixed.

Some of the job's histories may have been a bit red, that was due to some infra changes that took place last week.

I think all should be good here in general.

**Comment by luxas** [comment]
Ah, didn't see @pipejakob's comment, we basically say the same thing :wink:

**Comment by jdumars** [comment]
> I see two completely green runs on 9/22, followed by one that is still running. Is this an argument for adding the test to release-master-blocking? I thought we had gotten messaging from @jdumars during a SIG Cluster Lifecycle to keep the kubeadm e2e tests out of release-master-blocking, but that adding them to 1.8-blocking was still fine.

I believe the hope was to ensure it was passing/curated, even if not in release-master-blocking, which appears to have been the case.  I have been struggling to find the right balances with kubeadm, so forgive me if this has gone back and forth some.  

Bottom line, if the test is reliable and giving us solid quality signal for the integration post-release, I am fine with it being blocking because it hits some aspects that may not be covered in other use cases, as I believe we talked about some time ago.  My main concern at this point is that we're not trying to rush things in at the end, and we're not ignoring release quality data that could come back to haunt us later.  

I am of course open to disagreement whichever way the community feels most strongly.

**Comment by spiffxp** [comment]
@pipejakob @luxas ok, thanks for the feedback, between here and slack I have a better understanding

Given that gce-kubeadm-1.8 and periodic-gce-kubeadm-1.8 exercise the same code, how about we keep the higher-frequency periodic in -blocking, and drop the postsubmit?

IMO it's totally fine to move https://k8s-testgrid.appspot.com/sig-cluster-lifecycle#kubeadm-gce to https://k8s-testgrid.appspot.com/release-master-blocking.  My only hesitance or source of confusion is over which version of kubeadm it's testing, and which version of kubernetes.  But basically as long as it's (mostly) green, actively watched and maintained, and valuable signal for cutting alphas & betas of kubernetes/kubernetes, that's where I'd want it to be.

**Comment by ericchiang** [comment]
I would like to make the network partition upgrade tests as non-blocking for 1.8

https://github.com/kubernetes/kubernetes/issues/47820#issuecomment-332048833

**Comment by ericchiang** [comment]
It was also requested that the stateful set failures not be marked release blocking https://github.com/kubernetes/kubernetes/issues/52808#issuecomment-331311989

**Comment by spiffxp** [comment]
>I would like to make the network partition upgrade tests as non-blocking for 1.8

@ericchiang there is one network partition test I didn't see explicitly called out in #47820 that I haven't marked as non-blocking yet, just since I haven't yet had time to see if that fails for the same reasons as the others; it fails in some upgrade tests where the others don't show up
>[k8s.io] Network Partition [Disruptive] [Slow] [k8s.io] Pods should return to running and ready state after network partition is healed All pods on the unreachable node should be marked as NotReady upon the node turn NotReady AND all pods should be mark back to Ready when the node get back to Ready before pod eviction timeout

>It was also requested that the stateful set failures not be marked release blocking #52808 (comment)
done (I regret having done this as a github issue, a spreadsheet would've been better, oh well)

**Comment by jdumars** [comment]
@spiffxp live by GitHub.  Die by GitHub.

**Comment by spiffxp** [comment]
Thanks to https://github.com/kubernetes/test-infra/pull/4727 we can probably rely on kubectl-skew tests, I'll update the list of upgrade tests to include the serial tests and whether they're worth paying attention to

**Comment by spiffxp** [comment]
https://github.com/kubernetes/test-infra/pull/4749 fixed [#53061](https://github.com/kubernetes/kubernetes/issues/53061) which means kops-aws-1.8 is usable as a blocking job

**Comment by spiffxp** [comment]
tl;dr I think we have relatively good signal on the release-1.8-blocking board, though it may not be totally green by the time we cut; release-1.8-upgrade-skew board will require more human interpretation, but no wildly unexpected failures thus far

I'm going to try converting this into something easier for human consumption and updating, but too late for east coast consumption today.  It might have the form of a spreadsheet.

**Comment by jdumars** [comment]
Thanks @spiffxp for https://docs.google.com/spreadsheets/d/1gCm5s_lzoJdmvjMrfuapeI10sm7J0roiH9WnIKMlsWQ/edit#gid=0

**Comment by luxas** [comment]
I can comment on the kubeadm upgrade job:
https://k8s-testgrid.appspot.com/sig-cluster-lifecycle#kubeadm-gce-upgrade-1.7-1.8 is working as expected

Breakdown: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/periodic-kubernetes-e2e-kubeadm-gce-upgrade-1-7-1-8/61

`Master upgrade hpa-upgrade 16m14s` fails
Which is expected because heapster is not deployed

`Timeout waiting for service "service-test" to have a load balancer` fails
Which is expected because because we don't use a cloud provider like GCE (required for `Service.Spec.Type=LoadBalancer`)

`Failed waiting for pods to enter running: timed out waiting for the condition` fails
because no default StorageClass is deployed, so nothing can create PVs for StatefulSets

So I think we're good :+1:
Many thanks to @jessicaochen for doing the main portion of the upgrade e2e work
Also kudos to @pipejakob and @jamiehannaford for helping out on various tasks

**Comment by luxas** [comment]
@spiffxp We can close this now, right?

**Comment by luxas** [comment]
Closing @spiffxp @jdumars reopen if you disagree
